dotnet/src/Agents/UnitTests/AzureAI/Definition/AzureAIAgentFactoryTests.cs

```diff
@@ -0,0 +1,122 @@
+            "endpoint;subscription_id;resource_group_name;project_name",
+            new FakeTokenCredential(),
+            new AIProjectClientOptions()
+            { Transport = new HttpClientTransport(this._httpClient) });
+        builder.Services.AddSingleton<AIProjectClient>(client);
+        this._kernel = builder.Build();
+    }
+
+    /// <inheritdoc/>
+    public void Dispose()
+    {
+        GC.SuppressFinalize(this);
+        this._messageHandlerStub.Dispose();
+        this._httpClient.Dispose();
+    }
+
+    /// <summary>
+    /// Verify can create an instance of <see cref="Microsoft.SemanticKernel.Agents.Agent"/> using <see cref="AzureAIAgentFactory"/>
+    /// </summary>
+    [Fact]
+    public async Task VerifyCanCreateAzureAIAgentAsync()
+    {
+        // Arrange
+        AgentDefinition agentDefinition = new()
+        {
+            Type = AzureAIAgentFactory.AzureAIAgentType,
+            Name = "AzureAIAgent",
+            Description = "AzureAIAgent Description",
+            Instructions = "AzureAIAgent Instructions",
+            Model = new()
+            {
+                Id = "gpt-4o-mini"
+            },
+            Tools = [
+                new Microsoft.SemanticKernel.Agents.AgentToolDefinition()
+                {
+                    Id = "tool1",
+                    Type = "code_interpreter",
+                },
+            ]
+        };
+        AzureAIAgentFactory factory = new();
+        this.SetupResponse(HttpStatusCode.OK, AzureAIAgentResponse);
+
+        // Act
+        var agent = await factory.CreateAsync(this._kernel, agentDefinition);
+
+        // Assert
+        Assert.NotNull(agent);
+        Assert.Equal(agentDefinition.Name, agent.Name);
+        Assert.Equal(agentDefinition.Description, agent.Description);
+        Assert.Equal(agentDefinition.Instructions, agent.Instructions);
+        Assert.Equal(this._kernel, agent.Kernel);
+    }
+
+    /// <summary>
+    /// Azure AI Agent response.
+    /// </summary>
+    public const string AzureAIAgentResponse =
+        """
+        {
+          "id": "asst_thdyqg4yVC9ffeILVdEWLONT",
+          "object": "assistant",
+          "created_at": 1739991984,
+          "name": "AzureAIAgent",
+          "description": "AzureAIAgent Description",
+          "model": "gpt-4o",
+          "instructions": "AzureAIAgent Instructions",
+          "tools": [],
+          "top_p": 1.0,
+          "temperature": 1.0,
+          "tool_resources": {},
+          "metadata": {},
+          "response_format": "auto"
+        }
+        """;
+
+    #region private
+    private void SetupResponse(HttpStatusCode statusCode, string response) =>
+#pragma warning disable CA2000 // Dispose objects before losing scope
+        this._messageHandlerStub.ResponseQueue.Enqueue(new(statusCode)
+        {
+            Content = new StringContent(response)
+        });
+#pragma warning restore CA2000 // Dispose objects before losing scope
```

Comment:
Return the response in the method so you can `using` it without having to suppress a pragma.

```suggestion
    private HttpResponse SetupResponse(HttpStatusCode statusCode, string response) 
    {
        var response = new HttpReponse(statusCode)
        {
            Content = new StringContent(response)
        };
        
        this._messageHandlerStub.ResponseQueue.Enqueue(response);
        
        return response;
    }
```